File: Rotate Image/Rotate Image.java
public class Solution {
    public void rotate(int[][] matrix) {
        int rlen = matrix.length;
        int clen = matrix[0].length;
        if(rlen != clen){
            return;
        }
        int rowSize = 0;
        int colSize = 0;
        if(rlen % 2 == 1){
            rowSize = rlen / 2 + 1;
            colSize = rlen / 2;
        }
        else{
            rowSize = rlen / 2;
            colSize = rlen / 2;
        }
        for(int i = 0; i < rowSize; ++i){
            for(int j = 0; j < colSize; ++j){
                int temp = matrix[i][j];
                matrix[i][j] = matrix[clen - j - 1][i];
                matrix[clen - j - 1][i] = matrix[clen - i - 1][clen - j - 1];
                matrix[clen - i - 1][clen - j - 1] = matrix[j][clen - i - 1];
                matrix[j][clen - i - 1] = temp;
            }
        }
        return;
    }
}

File: Rotate Image/Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int> > &matrix) {
        int n = matrix.size();
        if(!n || (n == 1))  return;
        
        int row = 0;
        int colomn = 0;
        
        if(n % 2){
            row = n / 2 + 1;
            colomn = n / 2;
        }
        else{
            row = n / 2;
            colomn = n / 2;
        }
        for(unsigned int i = 0; i < row; ++i){
            for(unsigned int j = 0; j < colomn; ++j){
                int temp = matrix[i][j];
                matrix[i][j] = matrix[n - j - 1][i];
                matrix[n - j - 1][i] = matrix[n - i - 1][n - j - 1];
                matrix[n - i - 1][n - j - 1] = matrix[j][n - i - 1];          
                matrix[j][n - i - 1] = temp;
            }
        }
    }
};
